### DISCUSSION TITLE
Saving and restoring model while compiling with 2 optimizers

### DISCUSSION THREAD

**Discussion by KunalA18** [lululxvi/deepxde#1429] [Q&A] [answered]
Hi,  I wish to save my model and then in a different script restore it to make multiple predictions. 
Can someone help me with save and restore process as here I am compiling my model with 2 optimizers.

here's the code for training my model:
``` 
  model = dde.Model(data, net)
  model.compile("adam", lr=0.008)
  model.train(iterations=5000, display_every=500)
  model.compile("L-BFGS")
  losshistory, train_state = model.train(iterations=1000, display_every=1000)
  dde.saveplot(losshistory, train_state, issave=True, isplot=True)
```

**Comment by vl-dud** [comment]
Hi!
To save:
```
  model = dde.Model(data, net)
  model.compile("adam", lr=0.008)
  model.train(iterations=5000, display_every=500)
  model.compile("L-BFGS")
  losshistory, train_state = model.train(iterations=1000, display_every=1000, model_save_path="path/to/save/your_model")
  dde.saveplot(losshistory, train_state, issave=True, isplot=True)
```
After save, choose file *.ckpt* in *path/to/save*:
```
model.compile("L-BFGS")
model.restore("path/to/save/your_model-<iterations>.ckpt")
```
Check the section "How can I save a trained model and then load the model later?" in [FAQ](https://deepxde.readthedocs.io/en/latest/user/faq.html)

**Reply by KunalA18** [reply]
Hi, 
Don't I have to save and restore after the model is compiled with adam optimizer?
just save and restore at the end?

**Reply by vl-dud** [reply, marked as answer]
it's just saving weights of the network. So one save is enough.
